staging/src/k8s.io/apimachinery/pkg/util/sets/set.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sets
+
+import "sort"
+
+type Ordered interface {
+	int | int8 | int16 | int32 | int64 | uint | uint8 | uint16 | uint32 | uint64 | uintptr | float32 | float64 | string
```

Comment:
Hey @logicalhan , IMHO a generic implementation is pointless if we limit it to anything other than `comparable`, at that point we can simply codegen the 14 implementations and the result will be faster and just as useful.

I also looked into this some time ago and got stuck on the problem that `List` sorts. Today I realized that   we can use `reflect` to sort if the type is Ordered and otherwise just return an unsorted slice: https://github.com/kubernetes/apimachinery/compare/master...alvaroaleman:generics?expand=1

WDYT? If the approach makes sense to you I can rebase the above against k/k

Reply:
Very sorry for stomping in like this. Back when I originally looked at this in Feb I couldn't find a way to use `reflect` to sort if possible, because `ordered` does not exist as a property. Because of that I didn't PR it because I didn't think it adds value if we have to limit it. Today I was made aware that while `reflect` doesn't expose `Ordered`, the set of types that are `ordered` is known and somewhat limited, so by explicitly opting in all types that are ordered, the list-if-ordered can be done. We could probably merge your change first and my changes to make it more generic on top to keep attribution (assuming approves are ok with adding a generic implementation)?